### DISCUSSION TITLE
Create a client Nextjs page to store tokens in zustand

### DISCUSSION THREAD

**Discussion by arsekil** [pmndrs/zustand#2901] [General] [answered]
Hello eceryone,

My main reason to post this question, is that is it a good approach to create a client side page, that runs my Store's setter function to store tokens acquire on the server side? 
Honestly, I couldn't think of a better way, because I know that I can't use zustand on the server side.

Basically, I don't render anything but a loading skeleton on the client side page, but in the background it calls my store's setter function to save my secret tokens acquired by server actions.

What are you taughts on this?

**Comment by dbritto-dev** [comment]
@andrejmoltok you could create your store with tokens from server

**Reply by arsekil** [reply]
Does this mean that i can call the store setter function on the server to save the tokens?

**Reply by dbritto-dev** [reply, marked as answer]
@andrejmoltok no, you can create your store from the beginning with that data. Ref: https://zustand.docs.pmnd.rs/guides/nextjs#initializing-the-store
